torchdata/csrc/CMakeLists.txt

```diff
@@ -31,14 +28,22 @@ if(BUILD_S3)
 
   add_library(_torchdata MODULE ${EXTENSION_SOURCES})
 
-  target_include_directories(_torchdata PRIVATE ${PROJECT_SOURCE_DIR} ${Python_INCLUDE_DIR} ${pybind11_INCLUDE_DIRS})
+  add_dependencies(_torchdata aws-cpp-sdk)
```

Comment:
Yes, we need `add_dependencies` to force `aws-cpp-sdk` builds before `_torchdata`, so that all necessary dependencies exist when building `_torchdata`. Without the forced order, `_torchdata` may fail to build.
Tried locally on ubuntu and found out that this line is a requirement.